editor/editor_help.cpp

```diff
@@ -2935,7 +2948,65 @@ void EditorHelp::_gen_extensions_docs() {
 	}
 }
 
-void EditorHelp::generate_doc(bool p_use_cache) {
+void EditorHelp::_load_script_doc_cache() {
+	ERR_FAIL_COND_MSG(!ProjectSettings::get_singleton()->is_project_loaded(), "Error: cannot load script doc cache without a project.");
+	if (ResourceLoader::exists(get_script_doc_cache_full_path())) {
+		Ref<Resource> script_doc_cache_res = ResourceLoader::load(get_script_doc_cache_full_path());
+		if (script_doc_cache_res.is_valid()) {
+			Array classes = script_doc_cache_res->get_meta("classes", Array());
+			for (int i = 0; i < classes.size(); i++)
+				doc->add_doc(DocData::ClassDoc::from_dict(classes[i]));
+		}
+		return;
+	}
+
+	// Force compile all scripts, which should generate their documentation.
+	WARN_PRINT("Script documentation cache not found. Regenerating it may take a while.");
+	HashMap<String, bool> is_extension_script; // Caches ScriptServer results so it's not being queried for every file, only for every extension.
```

Comment:
This probably doesn't make performance that much better. I wonder whether it's worth the extra complication.